istioctl/pkg/option/option.go

```diff
@@ -0,0 +1,115 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package option
+
+import (
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/utils/pointer"
+)
+
+const (
+	FlagKubeConfig     = "kubeconfig"
+	FlagContext        = "context"
+	FlagNamespace      = "namespace"
+	FlagIstioNamespace = "istioNamespace"
+)
+
+type RootFlags struct {
+	kubeconfig     *string
+	configContext  *string
+	namespace      *string
+	istioNamespace *string
+
+	defaultNamespace string
+}
+
+func NewRootFlags() *RootFlags {
```

Comment:
nit: merge AddFlags and NewRootFlags, no need to ever have uninitialized RootFlags?